File: 150-essential/h-index/java/Solution1.java
class Solution {
    public int hIndex(int[] citations) {
        int hIndex = 0;
        int[] numberOfCitations = new int[citations.length + 1];

        for (int i = 0; i < citations.length; i++) {
            if (citations[i] > citations.length) {
                numberOfCitations[numberOfCitations.length - 1]++;
            } else {
                numberOfCitations[citations[i]]++;
            }
        }

        int counter = 0;
        for (int j = numberOfCitations.length - 1; j > 0; j--) {
            counter += numberOfCitations[j];
            if (counter >= j) {
                hIndex = j;
                break;
            }
        }

        return hIndex;
    }
}

File: 150-essential/h-index/c-plus/solution1.cpp
class Solution
{
public:
    int hIndex(vector<int> &citations)
    {
        int hIndex = 0;
        int numberOfPapers[citations.size() + 1];
        int counterOfPapers = 0;

        for (int i = 0; i < citations.size())
        {
            if (citations[i] >= citations.size())
            {
                numberOfPapers[citations.size()] = citations[i];
            }
            else
            {
                numberOfPapers[citations[i]] = citations[i];
            }
        }

        for (int j = numberOfPapers.size() - 1; j > 0; j--)
        {
            counterOfPapers += numberOfPapers[j];
            if (counterOfPapers >= j)
            {
                hIndex = j;
                break;
            }
        }
    }
    return hIndex;
};